tests/system/test_cluster/test_agent_info_sync/test_agent_info_sync.py

```diff
@@ -54,6 +56,15 @@ def register_agent():
     """Restart the removed agent to trigger auto-enrollment."""
     yield
     host_manager.get_host(deleted_agent).ansible('command', f'service wazuh-agent restart', check=False)
+    timeout = time() + time_to_agent_reconnect
+    command = f'{WAZUH_PATH}/bin/cluster_control -a | grep active | wc -l'
+    while True:
+        if int(host_manager.run_shell('wazuh-worker2', command)) == 4 or time() > timeout:
+            if int(host_manager.run_shell('wazuh-worker1', command)) == 4 or time() > timeout:
+                if int(host_manager.run_shell('wazuh-master', command)) == 4 or time() > timeout:
+                    break
```

Comment:
This code will pass even if the command doesn't return 4 because of the timeout. We are not checking if the agent is active, we are just checking the number of lines.